akka-remote/src/main/java/akka/remote/artery/compress/CountMinSketch.java

```diff
@@ -0,0 +1,338 @@
+   */
+  public long addAndEstimateCount(String item, long count) {
+    if (count < 0) {
+      // Actually for negative increments we'll need to use the median
+      // instead of minimum, and accuracy will suffer somewhat.
+      // Probably makes sense to add an "allow negative increments"
+      // parameter to constructor.
+      throw new IllegalArgumentException("Negative increments not implemented");
+    }
+    final int[] buckets = MurmurHash.hashBuckets(item, depth, width);
+    for (int i = 0; i < depth; ++i) {
+      table[i][buckets[i]] += count;
+    }
+    size += count;
+    return estimateCount(buckets);
+  }
+
+  public long size() {
+    return size;
+  }
+  
+  /**
+   * The estimate is correct within {@code 'epsilon' * (total item count)},
+   * with probability {@code confidence}.
+   */
+  public long estimateCount(long item) {
+    long res = Long.MAX_VALUE;
+    for (int i = 0; i < depth; ++i) {
+      res = Math.min(res, table[i][hash(item, i)]);
+    }
+    return res;
+  }
+
+  /**
+   * The estimate is correct within {@code 'epsilon' * (total item count)},
+   * with probability {@code confidence}.
+   */
+  public long estimateCount(String item) {
+    int[] buckets = MurmurHash.hashBuckets(item, depth, width);
+    return estimateCount(buckets);
+  }
+
+  /**
+   * The estimate is correct within {@code 'epsilon' * (total item count)},
+   * with probability {@code confidence}.
+   * 
+   * @param buckets the "indexes" of buckets from which we want to calculate the count
+   */
+  private long estimateCount(int[] buckets) {
+    long res = Long.MAX_VALUE;
+    for (int i = 0; i < depth; ++i) {
+      res = Math.min(res, table[i][buckets[i]]);
+    }
+    return res;
+  }
+  
+  /**
+   * This is copied from https://github.com/addthis/stream-lib/blob/master/src/main/java/com/clearspring/analytics/hash/MurmurHash.java
+   * Which is Apache V2 licensed.
+   * <p></p>
+   * This is a very fast, non-cryptographic hash suitable for general hash-based
+   * lookup. See http://murmurhash.googlepages.com/ for more details.
+   * <p/>
+   * <p>
+   * The C version of MurmurHash 2.0 found at that site was ported to Java by
+   * Andrzej Bialecki (ab at getopt org).
+   * </p>
+   */
+  private static class MurmurHash {
+  
+    public static int hash(Object o) {
+      if (o == null) {
+        return 0;
+      }
+      if (o instanceof Long) {
+        return hashLong((Long) o);
+      }
+      if (o instanceof Integer) {
+        return hashLong((Integer) o);
+      }
+      if (o instanceof Double) {
+        return hashLong(Double.doubleToRawLongBits((Double) o));
+      }
+      if (o instanceof Float) {
+        return hashLong(Float.floatToRawIntBits((Float) o));
+      }
+      if (o instanceof String) {
+        return hash(((String) o).getBytes());
+      }
+      if (o instanceof byte[]) {
+        return hash((byte[]) o);
+      }
+      return hash(o.toString());
```

Comment:
Hm, great idea – yes we do use it on ActorRef, indirectly (passed in here it would have been a String, but we can change that).
Question is if that's going to work hm... `hashCode` seems to be `uid`, which is just a random number, CMS heavily relies on even hashcode distribution, thus murmuring keys.

I'm not sure the change would be valid, WDYT @drewhk ?